dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -305,7 +298,10 @@ public int authenticate(Context context, String username, String password,
 			}
 
 			// Get the Shib supplied affiliation or use the default affiliation
-			List<String> affiliations = findMultipleAttributes(request, roleHeader);
+                        List<String> affiliations = new ArrayList<String>();
+                        if (context.getCurrentUser().getMetadata(roleHeader) != null) {
```

Comment:
I'm pretty sure, that in my environment the request was not set and so the roleHeader did not exist. So I got the roleHeader (which in my case is configured as the simple "ou" attribute) from the ePerson metadata, where this value is stored by the eperson.metadata option in authentication-shibboleth.cfg). I have to admit, that this is a kind of bad dependency, but its working fine for me while the original code does not.

I will try to propose a solution, which is working for me, but which does not affect the existing configurations.